### ISSUE TITLE
Cache results of discover schema for connectors with static scehmas

### ISSUE THREAD

**Issue by cgardens** [airbytehq/airbyte#9930]
## Tell us about the problem you're trying to solve
Just like we cache a specs for connectors, for APIs (with static schemas), we can do the same thing with the output of discover schema.

This is a bad idea for connectors with dynamic schemas, but most of our API connectors, they are static for a given version of the connectors. We could add a pretty simple convention to label connectors where this is the case and fetch them from a cache. If did this, we could remove one of the really slow parts of the setup experience for most of our connectors.

While this issue is working in the same problem space as this epic: https://github.com/airbytehq/airbyte/issues/9895, it likely makes sense to do both. They will work fine together and one does not block the other.

**Comment by cgardens** [comment]
@davinchia I don't think so. @malikdiarra worked on storing each version of a catalog that was "discovered". This issue is to handle the case when we know a schema will never change (or at least doesn't change dynamically, i.e. not a database). Honestly I'm not sure it is too valuable anymore though after Malik's change.

If we did want to do this, we would need some way to mark that a connector had a static schema.

**Comment by davinchia** [comment]
thanks for the explanation @cgardens . agree it's not as important anymore. will keep it open for now unless you think we can close it out
